src/cache/cache_las.c

```diff
@@ -572,8 +608,23 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
          */
         WT_ASSERT(session, modifies.size > 0);
         __wt_modify_vector_pop(&modifies, &upd);
-        /* The key didn't exist back then, which is globally visible. */
-        WT_ASSERT(session, upd->type == WT_UPDATE_STANDARD || upd->type == WT_UPDATE_TOMBSTONE);
+
+        /* If we popped a modify then it should be flagged as in the history store. */
+        if (upd->type == WT_UPDATE_MODIFY) {
+            WT_ASSERT(session, F_ISSET(upd, WT_UPDATE_HISTORY_STORE));
```

Comment:
If you update restore, apply modifies on top and then reconcile again, won't this happen?

Reply:
Currently it happens. But for eviction, we can free everything inserted into lookaside right after commit. Thus, in the future, it won't happen.

Reply:
Yeah but doesn't it happen now as of the current state of the branch?

Reply:
No, in current state of the branch, the first update on stack is either a standard update or a tombstone.